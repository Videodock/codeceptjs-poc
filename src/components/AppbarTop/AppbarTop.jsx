import React from 'react';
import * as PropTypes from 'prop-types';
import classNames from 'classnames';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import { makeStyles } from '@material-ui/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import images from '../../theme/images';
import styles from './AppbarTop.styles';

const useStyles = makeStyles(styles);

export default function AppbarTop ({ children, position, mobile }) {
  const { t }   = useTranslation('common');
  const classes = useStyles();

  return (
    <AppBar position={position}>
      {!mobile && (
        <Toolbar classes={{ dense: classes.dense }} variant="dense">
          <Link to="/">
            <img className={classes.logo} src={images.logo} alt={t('trompa_logo')} />
          </Link>
          {children}
        </Toolbar>
      )}
      {mobile && (
        <Toolbar classes={{ dense: classNames(classes.dense, classes.mobile) }} variant="dense">
          <div className={classes.hamburger}>
            {children}
          </div>
          <Link to="/">
            <img className={classes.logo} src={images.logo} alt={t('trompa_logo')} />
          </Link>
        </Toolbar>
      )}
    </AppBar>
  );
}

AppbarTop.propTypes = {
  position: PropTypes.oneOf(['absolute', 'fixed', 'relative', 'static', 'sticky']),
  mobile  : PropTypes.bool,
};

AppbarTop.defaultProps = {
  position: 'relative',
  mobile  : 'false',
};
