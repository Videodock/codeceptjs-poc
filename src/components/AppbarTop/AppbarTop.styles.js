import { createStyles } from '@material-ui/styles';

export default ({ typography, palette }) => createStyles({
  dense: {
    height        : 56,
    color         : palette.text.blue,
    justifyContent: 'space-between',
    alignItems    : 'center',
    fontFamily    : typography.fontFamilyOpenSans,
    '& img'       : {
      transform: 'translateY(10%)',
    },
  },
  logo: {
    width: 138,
  },
  mobile: {
    position      : 'relative',
    justifyContent: 'center',
  },
  hamburger: {
    position : 'absolute',
    left     : 16,
    marginTop: 2,
    '& svg'  : {
      color: palette.common.black,
    },
  },
});
