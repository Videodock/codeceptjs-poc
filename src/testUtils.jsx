import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
