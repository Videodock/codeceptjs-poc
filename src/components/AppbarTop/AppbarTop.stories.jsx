import React from 'react';
import { storiesOf } from '@storybook/react';
import AppbarTop from './AppbarTop';

storiesOf('AppbarTop', module)
  .add('basic', () => <AppbarTop />);
