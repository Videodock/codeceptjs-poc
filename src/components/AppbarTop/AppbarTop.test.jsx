import React from 'react';
import { render } from '../../testUtils';
import AppbarTop from './AppbarTop';

describe('<AppbarTop />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<AppbarTop />);

    expect(container.firstChild).toMatchSnapshot();
  });

  // add your unit tests here
});
