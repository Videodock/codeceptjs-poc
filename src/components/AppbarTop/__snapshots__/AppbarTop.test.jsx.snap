// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppbarTop /> renders and matches snapshot 1`] = `
<header
  class="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary MuiPaper-elevation4"
>
  <div
    class="MuiToolbar-root MuiToolbar-dense makeStyles-dense-1 makeStyles-mobile-3 MuiToolbar-gutters"
  >
    <div
      class="makeStyles-hamburger-4"
    />
    <a
      href="/"
    >
      <img
        alt="Trompa logo"
        class="makeStyles-logo-2"
        src="logo.png"
      />
    </a>
  </div>
</header>
`;
